test/it/group-rule-operations.js

```diff
@@ -95,8 +95,8 @@ describe('Group-Rule API tests', () => {
     expect(userInGroup).to.equal(false);
 
     // 5. Delete the group, user and group rule
-    await utils.cleanup(client, createdUser, createdGroup);
     await updatedRule.deactivate();
     await updatedRule.delete();
+    await utils.cleanup(client, createdUser, createdGroup);
```

Comment:
User  and group should be deleted after deleting the group rule. Or else, the group rule goes into an `INVALID` state, causing failures sometimes depending on the delete operation, which is async.